pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -507,6 +506,19 @@ func updateObservedGeneration(app *v1beta1.Application) {
 	}
 }
 
+// filterManagedFieldChangesUpdate filter resourceTracker update event by ignoring managedFields changes
+// For old k8s version like 1.18.5, the managedField could always update and cause infinite loop
+// this function helps filter those events and prevent infinite loop
+func filterManagedFieldChangesUpdate(e ctrlEvent.UpdateEvent) bool {
+	new, isNewRT := e.ObjectNew.DeepCopyObject().(*v1beta1.ResourceTracker)
+	old, isOldRT := e.ObjectOld.DeepCopyObject().(*v1beta1.ResourceTracker)
+	if !isNewRT || !isOldRT {
+		return true
+	}
```

Comment:
seems these are duplicate code？

Reply:
Not exactly. One is to check type Application, here is to check type RT.